From 1e9a4afdfa995e924f74139f620e8b2f985a705f Mon Sep 17 00:00:00 2001
From: James Feist <james.feist@linux.intel.com>
Date: Tue, 10 Dec 2019 16:21:24 -0800
Subject: [PATCH 1/1] AST2500 increase boot speed

This hardcodes the IOMODE to x2 as that is the fastest
we support.

Signed-off-by: James Feist <james.feist@linux.intel.com>
---
 arch/arm/mach-aspeed/flash.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-aspeed/flash.c b/arch/arm/mach-aspeed/flash.c
index d33fb9e0fe..31d2ef6001 100644
--- a/arch/arm/mach-aspeed/flash.c
+++ b/arch/arm/mach-aspeed/flash.c
@@ -730,7 +730,7 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 	ulID = ((ulong)ch[0]) | ((ulong)ch[1] << 8) | ((ulong)ch[2] << 16) ;
 	info->flash_id = ulID;
 
-//	printf("SPI Flash ID: %x \n", ulID);
+	printf("SPI Flash ID: %x \n", ulID);
 
 	/* init default */
 	info->iomode = IOMODEx1;
@@ -1258,6 +1258,10 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 			} /* JDEC */
 	}
 
+	info->readcmd = 0xbb;
+	info->dualport = 1;
+	info->iomode = IOMODEx2_dummy;
+
 	sector = base;
 	for (j = 0; j < info->sector_count; j++) {
 
-- 
2.17.1

